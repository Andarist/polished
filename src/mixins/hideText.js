// @flow
import messageHandlers from '../internalHelpers/_messageHandlers'

/**
 * CSS to hide text to show a background image in a SEO-friendly way.
 *
 * @example
 * // Styles as object usage
 * const styles = {
 *   'backgroundImage': 'url(logo.png)',
 *   ...hideText(),
 * }
 *
 * // styled-components usage
 * const div = styled.div`
 *   backgroundImage: url(logo.png);
 *   ${hideText()};
 * `
 *
 * // CSS as JS Output
 *
 * 'div': {
 *   'backgroundImage': 'url(logo.png)',
 *   'textIndent': '101%',
 *   'overflow': 'hidden',
 *   'whiteSpace': 'nowrap',
 * }
 */

function hideText() {
  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production') {
    if (
      messageHandlers('mixins/hideText.js', {
        // eslint-disable-next-line prefer-rest-params
        arrityCheck: { args: arguments, max: 0 },
      })
    ) {
      return {}
    }
  }

  return {
    textIndent: '101%',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
  }
}

export default hideText
