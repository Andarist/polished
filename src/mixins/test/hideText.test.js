// @flow
import hideText from '../hideText'

describe('hideText', () => {
  beforeAll(() => {
    global.console = {
      error: jest.fn(),
      warn: jest.fn(),
    }
  })

  afterEach(() => {
    // eslint-disable-next-line no-console
    console.error.mockClear()
    // eslint-disable-next-line no-console
    console.warn.mockClear()
  })

  it('should return the CSS in JS', () => {
    expect({ ...hideText() }).toMatchSnapshot()
  })

  it('should add rules when block has existing rules', () => {
    expect({
      'background-image': 'url(logo.png)',
      ...hideText(),
    }).toMatchSnapshot()
  })

  it('should throw a warning when called with any parameters', () => {
    hideText('100%')
    // eslint-disable-next-line no-console
    expect(console.warn.mock.calls).toMatchSnapshot()
  })
})
