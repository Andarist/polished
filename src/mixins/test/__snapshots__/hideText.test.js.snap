// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hideText should return the CSS in JS 1`] = `
Object {
  "overflow": "hidden",
  "textIndent": "101%",
  "whiteSpace": "nowrap",
}
`;

exports[`hideText should throw a warning when called with any parameters 1`] = `Array []`;
